features/storage/kvstore/global_api/kvstore_global_api.cpp

```diff
@@ -0,0 +1,165 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "kvstore_global_api.h"
+
+#include "kv_config.h"
+#include "kv_map.h"
+#include "KVStore.h"
+#include "mbed_error.h"
+
+using namespace mbed;
+
+// iterator handle
+struct _opaque_kv_set_handle {
+    bool handle_is_open;
+    KVStore *kvstore_intance;
+    KVStore::set_handle_t *set_handle;
+};
+
+// iterator handle
+struct _opaque_kv_key_iterator {
+    bool iterator_is_open;
+    KVStore *kvstore_intance;
+    KVStore::iterator_t *iterator_handle;
+};
+
+int kv_set(const char *full_name_key, const void *buffer, size_t size, uint32_t create_flags)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_name_key, &kv_instance, key);
+    ret = kv_instance->set(key, buffer, size, create_flags);
+    return ret;
+}
+
+int kv_get(const char *full_name_key, void *buffer, size_t buffer_size, size_t *actual_size)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_name_key, &kv_instance, key);
+    return kv_instance->get(key, buffer, buffer_size, actual_size);
+}
+
+int kv_get_info(const char *full_name_key, kv_info_t *info)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_name_key, &kv_instance, key);
+    KVStore::info_t inner_info;
+    ret =  kv_instance->get_info(key, &inner_info);
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+    info->flags = inner_info.flags;
+    info->size =  inner_info.size;
+    return ret;
+}
+
+int kv_remove(const char *full_name_key)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_name_key, &kv_instance, key);
+    return kv_instance->remove(key);
+}
+
+int kv_iterator_open(kv_iterator_t *it, const char *full_prefix)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    (*it) = new _opaque_kv_key_iterator;
+    if (*it == NULL) {
+        return MBED_ERROR_FAILED_OPERATION;
+    }
+    (*it)->iterator_is_open = false;
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_prefix, &kv_instance, key);
+    (*it)->kvstore_intance = kv_instance;
+
+    KVStore::iterator_t *inner_it = new KVStore::iterator_t;
+    ret = kv_instance->iterator_open(inner_it, key);
+    if (MBED_SUCCESS != ret) {
+        delete inner_it;
+        return ret;
```

Comment:
*it was not deleted, and I think it wont be deleted in kv_iterator_close() because iterator_is_open = false